File: Spring/ProxiBanqueSI_JMH_JBB_SJ/src/main/java/org/proxibanque/model/CompteEpargne.java
package org.proxibanque.model;

import javax.persistence.Entity;

/**
 * Classe représentant les comptes Epargnes de la société Proxibanque
 * 
 * @author Jean-Michel HILTBRUNNER, Jean-Baptiste BLANC, Sebastien JOUMARD
 *
 */
@Entity
public class CompteEpargne extends Compte {

	private double taux = 0.03;

	// *** Constructor ***
	public CompteEpargne() {
		super();
	}

	// *** Getters & Setters
	public double getTaux() {
		return taux;
	}

	public void setTaux(double taux) {
		this.taux = taux;
	}

	// *** Methods ***
	@Override
	public String toString() {
		return " [" + taux + "]";
	}

}

File: Proxibanque-Angular-JBB-SJ-JMH/src/app/model/compteepargne.ts
import { Compte } from "./compte";

export class CompteEpargne extends Compte{
    id: number;
    numero: number;
    solde: number;
    date: string;
    taux: number = 0.03;

    constructor(options: {
        id: number;
        numero: number;
        solde: number;
        date: string;
        taux: number;
    }) {
        super(options);
        this.taux = options.taux;
    }

}